File: retbotbackend/src/main/java/dev258/retbotbackend/publicacao/entity/Publicacao.java
package dev258.retbotbackend.publicacao.entity;

import dev258.retbotbackend.utilizador.entity.ContaSocial;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.OffsetDateTime;

@Entity
@Table(name = "publicacao")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Publicacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_publicacao")
    private Long idPublicacao;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_conta_social", nullable = false)
    private ContaSocial contaSocial;

    @Column(name = "id_publicacao_externa", nullable = false, unique = true, length = 80)
    private String idPublicacaoExterna;

    @Column(name = "texto", columnDefinition = "TEXT")
    private String texto;

    @Column(name = "publicado_em")
    private OffsetDateTime publicadoEm;

    @Column(name = "criado_em", nullable = false, updatable = false)
    private OffsetDateTime criadoEm;

    @PrePersist
    protected void aoPersistir() {
        if (criadoEm == null) {
            criadoEm = OffsetDateTime.now();
        }
    }
}

File: retbot-frontend/src/models/Publicacao.js
export default class Publicacao {
  constructor({
    idPublicacao,
    idContaSocial,
    idPublicacaoExterna,
    texto,
    publicadoEm,
    criadoEm
  }) {
    this.idPublicacao = idPublicacao;
    this.idContaSocial = idContaSocial;
    this.idPublicacaoExterna = idPublicacaoExterna;
    this.texto = texto;
    this.publicadoEm = publicadoEm ? new Date(publicadoEm) : null;
    this.criadoEm = criadoEm ? new Date(criadoEm) : null;
  }
}